### PR TITLE
feat(theme): add LoadWhenVisible component and lazyload footer

### PR DESCRIPTION
**Pull request by bartoszherba**
## Description
- update social icons to reduce weight and increase the quality
- load footer only when visible

## Related Issue
-

## Motivation and Context
Refactoring and performance improvement

## How Has This Been Tested?
Observe networking if the footer is loaded only when visible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/components/LoadWhenVisible.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export default defineComponent({
+  name: 'LoadWhenVisible',
+  setup() {
+    const isVisible = ref(false);
+    const root = ref(null);
+
+    let observer = null;
+    if (process.browser && ('IntersectionObserver' in window)) {
+      observer = new IntersectionObserver((entries) => {
```
Comment (Frodigo) [inline packages/theme/components/LoadWhenVisible.vue:19]: Could you add the `rootMargin` prop and bind it to IntersectionObserver initialization?

File: packages/theme/layouts/default.vue [hunk 2]
```diff
@@ -14,7 +14,9 @@
       <nuxt :key="route.fullPath" />
     </div>
     <BottomNavigation />
-    <AppFooter />
+    <LoadWhenVisible>
```
Comment (Frodigo) [inline packages/theme/layouts/default.vue:17]: When you add `rootMargin` option, please set up it here. Please try `100px 0 0 0`.

Thanks to that footer should be shown a little bit earlier, and UX will be better.

**Review by Frodigo — CHANGES_REQUESTED** [review]

File: packages/theme/components/LoadWhenVisible.vue (hunk 1 above)
Comment (bartoszherba) [reply packages/theme/components/LoadWhenVisible.vue]: cool, thx

File: packages/theme/layouts/default.vue (hunk 2 above)
Comment (bartoszherba) [reply packages/theme/layouts/default.vue:17]: Oki

**Review by Frodigo — APPROVED** [review]

**Pull request merged**